src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs

```diff
@@ -166,45 +169,69 @@ public PublicClientApplicationBuilder WithBroker(bool enableBroker = true)
+                    "\n\rFor details see https://aka.ms/msal-net-wam ");
             }
 #endif
 
-#if NET_CORE
-            if (Config.BrokerCreatorFunc == null && DesktopOsHelper.IsWindows())
+#if NET_CORE && !NET6_WIN
+            if (Config.BrokerCreatorFunc == null)
             {
                 throw new PlatformNotSupportedException(
-                    "If you have a .NET Core 3.1 application, install the NuGet package Microsoft.Identity.Client.Desktop and call the extension method .WithWindowsBroker() first. " +
-
-                    "\n\r\n\rIf you have a Windows application which targets net5.0, net5.0-windows, net6.0, or net6.0-windows, change the target to at least net6.0-windows10.0.17763.0. \nYour app can still run on earlier versions of Windows such as Windows 7 if you add <SupportedOSPlatformVersion>7</SupportedOSPlatformVersion> in the csproj.\n The Windows broker (WAM) is available only on Windows 10+ and Windows Server 2019+; this library will fallback to a browser on older systems. " +
-
-                    "\n\r\n\rIf you have a .NET 6 cross-platform (Windows, Mac, Linux) application, dual target net6.0 and net6.0-windows10.0.17763.0. Your installer should deploy the net6.0 version on Mac and Linux and the net6.0-window10.0.17763.0 on Windows." +
-
-                    "\n\r\n\rIf you want to try the new broker preview, install the NuGet package Microsoft.Identity.Client.Broker and call the extension method .WithBrokerPreview(). " +
-                    "\n\rFor details, see https://aka.ms/msal-net-wam and https://github.com/dotnet/designs/blob/main/accepted/2020/platform-checks/platform-checks.md ");
+                    "The desktop broker is not directly available in the MSAL package. "+
+                    "\n\rInstall the NuGet package Microsoft.Identity.Client.Desktop and call the extension method .WithBroker(BrokerOptions). " +
+                    "\n\rFor details, see https://aka.ms/msal-net-wam");
             }
 #endif
 
+#if NET6_WIN
+            Config.BrokerOptions = new BrokerOptions(BrokerOptions.OperatingSystems.Windows);
+#endif
+
             Config.IsBrokerEnabled = enableBroker;
             return this;
 #pragma warning restore CS0162 // Unreachable code detected
         }
 
         /// <summary>
-        /// Allows customization of the Windows 10 Broker experience
+        /// Allows customization of the Windows 10 Broker experience. 
         /// </summary>
-#if !SUPPORTS_BROKER || __MOBILE__
+#if !NET6_WIN && !WINDOWS_APP
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("This API has been replaced with WithBroker(BrokerOptions), which can be found in Microsoft.Identity.Client.Broker package. See https://aka.ms/msal-net-wam for details.", false)]
+#endif
+#if NET6_WIN
         [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("This API has been replaced with WithBroker(BrokerOptions). See https://aka.ms/msal-net-wam for details.", false)]

```

Comment:
```suggestion
        [Obsolete("This API has been replaced with .WithBroker(BrokerOptions). See https://aka.ms/msal-net-wam for details.", false)]

```

Reply:
I disagree with having '.' before the method name in the comment.